File: src/main/java/frc/robot/subsystems/indexer/IndexerIO.java
package frc.robot.subsystems.indexer;

import org.littletonrobotics.junction.AutoLog;

public interface IndexerIO {
    
    @AutoLog
    public static class IndexerIOInputs {
        /**
         * Indexer wheel velocity in rotations/second
         */

        public double indexerVelocity = 0.0;

        /**
         * Indexer motor current in amperes
         */

        public double indexerCurrentAmps = 0.0;

        /**
         * Indexer motor applied voltage in volts
         */

        public double indexerAppliedVoltage = 0.0;

        /**
         * Indexer motor temperature in celcius
         */

        public double indexerTempCelcius = 0.0;
        /**
         * Indexer wheel velocity in rotations/second
         */

        public double feederVelocity = 0.0;

        /**
         * Indexer motor current in amperes
         */

        public double feederCurrentAmps = 0.0;

        /**
         * Indexer motor applied voltage in volts
         */

        public double feederAppliedVoltage = 0.0;

        /**
         * Indexer motor temperature in celcius
         */

        public double feederTempCelcius = 0.0;
    }

    public default void updateInputs(IndexerIOInputs inputs) {}

    public default void setIndexerSpeed(double speed) {}

    public default void stopIndexer() {}
    
    public default void setIdleMode(boolean isBrake) {}

    public default void periodic() {}
}

File: src/main/java/frc/robot/subsystems/indexer/IndexerIO.kt
package frc.robot.subsystems.indexer

import com.lobstahbots.junction.AutoLogKt
import com.lobstahbots.units.*

interface IndexerIO {
    @AutoLogKt
    open class IndexerIOInputs {
        /**
         * Indexer wheel velocity in rotations/second
         */
        var indexerVelocity = 0.rotationsPerSecond

        /**
         * Indexer motor current in amperes
         */
        var indexerCurrent = 0.amps

        /**
         * Indexer motor applied voltage in volts
         */
        var indexerAppliedVoltage = 0.volts

        /**
         * Indexer motor temperature in celcius
         */
        var indexerTemp = 0.celsius

        /**
         * Indexer wheel velocity in rotations/second
         */
        var feederVelocity = 0.rotationsPerSecond

        /**
         * Indexer motor current in amperes
         */
        var feederCurrent = 0.amps

        /**
         * Indexer motor applied voltage in volts
         */
        var feederAppliedVoltage = 0.volts

        /**
         * Indexer motor temperature
         */
        var feederTemp = 0.celsius
    }

    fun updateInputs(inputs: IndexerIOInputs) {}

    fun setIndexerSpeed(speed: Double) {}

    fun setFeederSpeed(speed: Double) {}

    fun stopIndexer() {}

    fun setIdleMode(isBrake: Boolean) {}

    fun periodic() {}
}
